storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.scheduler;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.storm.utils.Utils;
+
+public class DefaultScheduler implements IScheduler {
+
+    private static Set<WorkerSlot> badSlots(Map<WorkerSlot, List<ExecutorDetails>> existingSlots, int numExecutors, int numWorkers) {
+        if (numWorkers != 0) {
+            Map<Integer, Integer> distribution = Utils.integerDivided(numExecutors, numWorkers);
+            Set<WorkerSlot> _slots = new HashSet<WorkerSlot>();
+
+            for (Entry<WorkerSlot, List<ExecutorDetails>> entry : existingSlots.entrySet()) {
+                Integer executorCount = distribution.get(entry.getValue().size());
```

Comment:
the value returned is not actually the executor count but the number of workers with executor count = entry.getValue().size(). I would prefer to refactor it 
```
Integer executorCount = entry.getValue().size();
Integer workerCount = distribution.get(executorCount);
if (workerCount != null && workerCount > 0) {
                     _slots.add(entry.getKey());
                     workerCount--;
                     distribution.put(executorCount, workerCount);
                 }
```